modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/serviceextensions/preupdatepassword/PreUpdatePasswordActionFailureTestCase.java

```diff
@@ -242,6 +243,43 @@ public void testAdminUpdatePassword() throws Exception {
 //     }
 
     @Test(dependsOnMethods = "testAdminUpdatePassword",
+            description = "Verify the user password recovery flow failure with pre update password action")
+    public void testUserResetPasswordWithPasswordRecoveryFlow() throws Exception {
+
+        updateFlowStatus(PASSWORD_RECOVERY_FLOW_TYPE, true);
+        addPasswordRecoveryFlow(flowManagementClient);
+        flowExecutionClient.initiateFlowExecution(PASSWORD_RECOVERY_FLOW_TYPE);
+
+        FlowExecutionRequest step1Request = buildPasswordRecoveryFlowRequest();
+        FlowExecutionResponse step1Response =
+                (FlowExecutionResponse) flowExecutionClient.executeFlow(step1Request);
+
+        String otpCode = getOTPFromEmail();
+        FlowExecutionRequest step2Request = buildOTPVerificationRequest(step1Response.getFlowId(), otpCode);
+        FlowExecutionResponse step2Response =
+                (FlowExecutionResponse) flowExecutionClient.executeFlow(step2Request);
+
+        FlowExecutionRequest step3Request = buildPasswordResetRequest(step2Response.getFlowId());
+        Object step3ResponseObj = flowExecutionClient.executeFlow(step3Request);
+        assertTrue(step3ResponseObj instanceof Error, "Unexpected response type for flow execution.");
```

Comment:
In these test context, we only require the flow execution response of the last step. 
Hence we should not have all the steps in the test case.

We can move all the steps to a method in the TestBase class.

ideally we should have something like this in the test class.

```
Object flowExecutionResponse = executePasswordRecoveryFlow(username, password);
```